src/renderers/dom/shared/CSSPropertyOperations.js

```diff
@@ -106,6 +120,10 @@ if (__DEV__) {
     } else if (badStyleValueWithSemicolonPattern.test(value)) {
       warnStyleValueWithSemicolon(name, value);
     }
+
+    if (numberIsNaN(value)) {
```

Comment:
The advantage to not inlining is readability, since otherwise `if(value !== value)` has an incredibly obscure behavior.  I don't feel strongly, so if you want it inlined, that's fine with me.